snooty/semanticparser.py

```diff
@@ -8,21 +8,44 @@
 class SemanticParser:
     def __init__(self, project_config: ProjectConfig) -> None:
         self.project_config = project_config
+        self.slug_title: Dict[str, Any] = {}
+        self.toctree: Dict[str, Any] = {}
+        self.pages: Dict[str, Any] = {}
 
-    def run(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
+    def run(
+        self, pages: Dict[FileId, Page], fn_names: List[str]
+    ) -> Dict[str, SerializableType]:
         # Specify which transformations should be included in semantic postprocessing
-        functions: List[Callable[[Dict[FileId, Page]], Dict[str, SerializableType]]] = [
-            self.toctree,
-            self.slug_title,
-        ]
+
+        functions: List[
+            Callable[[Dict[FileId, Page]], Dict[str, SerializableType]]
+        ] = self.functions(fn_names)
+
         document: Dict[str, SerializableType] = {}
 
         for fn in functions:
             field: Dict[str, SerializableType] = fn(pages)
             document.update(field)
         return document
 
-    def slug_title(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
+    # Returns a list of transformations to include in self.run()
+    def functions(
+        self, fn_names: List[str]
+    ) -> List[Callable[[Dict[FileId, Page]], Dict[str, SerializableType]]]:
+        fn_list: List[Any] = []
+        for name in fn_names:
+            if name == "toctree":
+                fn_list.append(self.build_toctree)
+            elif name == "slug-title":
+                fn_list.append(self.build_slug_title)
+            elif name == "breadcrumbs":
+                fn_list.append(self.breadcrumbs)
```

Comment:
I think a string-to-function mapping may be a bit simpler than this conditional statement. Something like this:
```python
fn_mapping = {
    "toctree": self.build_toctree
    # etc.
}
```

Reply:
good idea!!